### PR TITLE
Create a code fix that offers to add a reference to System.ValueTuple if it is missing and you use a tuple.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This is a port of https://github.com/dotnet/roslyn/pull/15014 to RC2.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide @Pilchie

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz @natidea .  Template coming shortly.

**Comment by CyrusNajmabadi** [comment]
o	Customer scenario –

Customer tries to use tuples.  Because they don't have the appropriate nuget package, the compiler immediately reports an error.  With this change, if they ahve enabled 'offer nuget package', then they'll get a fix that will add "System.ValueTuple" to their project.

o	Bugs this fixes:  (either VSO or GitHub links)

https://github.com/dotnet/roslyn/issues/13375

o	Workarounds, if any – and why we think they are insufficient for RC vs. RC2, RC3, or RTW

User can manually add the package themselves.  We think this is a poor experience.

o	Risk – this is generally a measure our how central the affected code is to adjacent scenarios and thus how likely your fix is to destabilize a broader area of code

This code runs automatically during editing, if the user uses a tuple.  So it's medium risk.

o	Performance impact (with a brief justification for that assessment – e.g. “Low perf impact because no extra allocations/no complexity changes” vs. “Low”)

Perf impact should be low.  The code should only run when the compiler produces a very specific error. And this is all through the same codepath that we use to process all errors in order to report fixes.  That code is very well covered, and we should not have any perf issues here.

o	Is this a regression from a previous update?

No.

o	Root cause analysis:  how did we miss it?  What tests are we adding to guard against it in the future?

This is not a regression.  It is new feature work.

o	How was the bug found?  (E.g. customer reported it vs. ad hoc testing)

Known as part of the ValueTuple feature.   It's just come along very late due to many moving parts across many teams.

**Comment by MattGertz** [comment]
Approved pending the remaining CR (there is one in the linked issue already.

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide I need a second CR for this since this is for ask-mode.  Thanks!

**Comment by CyrusNajmabadi** [comment]
Ping @dotnet/roslyn-ide I need a second PR on this ASAP.  This is for RC2.

File: src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngine.cs [hunk 1]
```diff
@@ -108,7 +108,25 @@ public SymbolSearchUpdateEngine(ISymbolSearchLogService logService)
...
         }
 
-        public Task<ImmutableArray<PackageWithAssemblyResult>> FindPackagesWithAssemblyAsync(
+        public async Task<ImmutableArray<PackageWithAssemblyResult>> FindPackagesWithAssemblyAsync(
+            string source, string assemblyName)
+        {
+            var result = await FindPackagesWithAssemblyWorkerAsync(source, assemblyName).ConfigureAwait(false);
+
+#if DEBUG
+            // For testing purposes, we hardcode this in if we're in DEBUG mode.
```
Comment (dpoeschl) [inline src/Workspaces/Core/Desktop/SymbolSearch/SymbolSearchUpdateEngine.cs]: What about when I'm in debug but not running tests? Should there be a better test hook here?

**Pull request merged**